hail/src/main/scala/is/hail/types/encoded/ETransposedArrayOfStructs.scala

```diff
@@ -158,57 +158,42 @@ final case class ETransposedArrayOfStructs(
     )
   }
 
-  def _buildSkip(mb: EmitMethodBuilder[_], r: Value[Region], in: Value[InputBuffer]): Code[Unit] = {
-    val len = mb.newLocal[Int]("len")
-    val i = mb.newLocal[Int]("i")
-    val nMissing = mb.newLocal[Int]("nMissing")
-    val alen = mb.newLocal[Int]("alen")
-    val anMissing = mb.newLocal[Int]("anMissing")
-    val fmbytes = mb.newLocal[Long]("fmbytes")
-    val skipFields = Code(fields.map { f =>
-      val skip = f.typ.buildSkip(mb)
-      if (f.typ.required) {
-        Code(
-          i := 0,
-          Code.whileLoop(i < alen, Code(skip(r, in), i := i + 1)))
-      } else {
-        Code(
-          in.readBytes(r, fmbytes, anMissing),
-          i := 0,
-          Code.whileLoop(i < alen,
-            Code(
-              Region.loadBit(fmbytes, i.toL).mux(
-                Code._empty,
-                skip(r, in)),
-              i := i + 1)))
-      }
-    }.toArray)
+  def _buildSkip(cb: EmitCodeBuilder, r: Value[Region], in: Value[InputBuffer]): Unit = {
+    val len = cb.newLocal[Int]("len")
+    val i = cb.newLocal[Int]("i")
+    val nMissing = cb.newLocal[Int]("nMissing")
+    val alen = cb.newLocal[Int]("alen")
+    val anMissing = cb.newLocal[Int]("anMissing")
+    val fmbytes = cb.newLocal[Long]("fmbytes")
 
+    cb.assign(len, in.readInt())
+    cb.assign(nMissing, UnsafeUtils.packBitsToBytes(len))
     val prefix = Code(len := in.readInt(), nMissing := UnsafeUtils.packBitsToBytes(len))
     if (structRequired) {
-      Code(
-        prefix,
-        alen := len,
-        anMissing := UnsafeUtils.packBitsToBytes(alen),
-        fmbytes := r.allocate(const(1), nMissing.toL),
-        skipFields
-      )
+      cb.assign(alen, len)
```

Comment:
Previous version had `anMissing := UnsafeUtils.packBitsToBytes(alen)` here. Was that removed intentionally? It looks to me like that could be factored out of the if branches to the `fields.foreach`. And the `fmbytes` assignment as well.